test/lib/ansible_test/_data/sanity/code-smell/runtime-metadata.py

```diff
@@ -68,7 +94,34 @@ def any_value(value):
     return value
 
 
-def validate_metadata_file(path, is_ansible):
+def get_ansible_version():
+    """Return current ansible-core version"""
+    from ansible.release import __version__
+
+    return LooseVersion('.'.join(__version__.split('.')[:3]))
+
+
+def get_collection_version():
+    """Return current collection version, or None if it is not available"""
+    import importlib.util
```

Comment:
Should this handle Python 2 as well? It looks like other testing code does this https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_internal/util.py#L812-L835

Reply:
Indeed. I assumed that all code-smell tests require Python 3.6+, but that only happens when `minimum_python_version` is specified in the associated JSON file. I'll fix that later today.

Reply:
Thanks, sorry for not catching that in the initial review.

Reply:
After digging some more, it looks like this is not necessary: all code-smell tests inherit `supported_python_versions` from `SanityTest`, which is defined as all supported Python versions >= 3.0: https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_internal/sanity/__init__.py#L658-L661

If `minimum_python_version` is specified for a code-smell test, it can have even stricter requirements (https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_internal/sanity/__init__.py#L788), but no code-smell test will ever work with Python 2.x (or in fact Python 3.4 or earlier).